transaction/transaction.go

```diff
@@ -639,6 +643,120 @@ func (tx *Transaction) HashForWitnessV0(inIndex int, prevoutScript []byte, value
 	return chainhash.DoubleHashH(s.Bytes())
 }
 
+// HashForWitnessV1 returns the sighash as described in BIP-340
+func (tx *Transaction) HashForWitnessV1(
+	inIndex int,
+	prevoutsScripts [][]byte,
+	prevoutsAssetValues []struct {
+		Asset []byte
+		Value []byte
+	},
```

Comment:
could it make sense to use a list of prevouts here like `prevouts Output[]` instead of these args?

Otherwise, iIwould go for `prevoutsScripts`, `prevoutsAssets` and `prevoutsValues` ... not so common to have untyped structs as func args.